src/internal/utils.js

```diff
@@ -19,6 +19,11 @@ export function check(value, predicate, error) {
   }
 }
 
+const hasOwnProperty = Object.prototype.hasOwnProperty
+export function hasProperty(object, property) {
```

Comment:
dont worry about this ordering, circular dependencies are fun ;)

though I feel like `hasOwn` would be better name for this utility, from `hasProperty` somebody could read that it should return `true` also for inherited ones